airbyte-integrations/connectors/source-kafka/src/main/java/io/airbyte/integrations/source/kafka/generator/GeneratorFactory.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.kafka.generator;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.source.kafka.KafkaConsumerRebalanceListener;
+import io.airbyte.integrations.source.kafka.KafkaProtocol;
+import io.airbyte.integrations.source.kafka.KafkaStrategy;
+import io.airbyte.integrations.source.kafka.MessageFormat;
+import io.airbyte.integrations.source.kafka.converter.AvroConverter;
+import io.airbyte.integrations.source.kafka.converter.Converter;
+import io.airbyte.integrations.source.kafka.converter.JsonConverter;
+import io.airbyte.integrations.source.kafka.mediator.KafkaMediator;
+import io.airbyte.integrations.source.kafka.mediator.KafkaMediatorImpl;
+import io.airbyte.integrations.source.kafka.state.StateHelper;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClientConfig;
+import io.confluent.kafka.serializers.KafkaAvroDeserializer;
+import io.confluent.kafka.serializers.KafkaAvroDeserializerConfig;
+import io.confluent.kafka.serializers.KafkaAvroSerializerConfig;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.connect.json.JsonDeserializer;
+
+public class GeneratorFactory {

```

Comment:
You mean, not passing json as input but some variables that let it decide what to create? Yes, I agree. I always review my own PRs when I create them, and this bit is something I don't quite like and I was thinking about a similar refactoring. Instead of passing the config and state as JSON pass some arguments - format, maxRecords, nrOfRetries and the actual state. Is that what you mean?

Reply:
Exactly 👍🏻 , in that way the Factory won't be aware of Json details that they are not part of the responsibilities the Factory owns.